src/test/clojure/clara/test_durability.clj

```diff
@@ -1,163 +1,257 @@
+                                      "expected fact: " expected-fact \newline
+                                      "fact: " fact \newline))
+                            fact))
+
+               :else
+               (is false
+                   (str "Must find a matching comparison with the expected."
+                        "  Most of the time this means the facts should be identical?" \newline
+                        "expected fact: " expected-fact \newline
+                        "fact: " fact \newline)))))))
+
+(defn durability-test
+  "Test runner to run different implementations of d/ISessionSerializer."
+  [serde-type]
+  (let [s (mk-session 'clara.durability-rules)
+
+        ;; Testing identity relationships on the IWorkingMemorySerializer facts received to serialize.
+        ;; So this is a little weird, but we want to know the exact object identity of even these
+        ;; "primitive" values.
+        mci "MCI"
+        lax "LAX"
+        san "SAN"
+        chi "CHI"
+        irk "IRK"
+        ten 10
+        twenty 20 
+        fifty 50
+        forty 40
+        thirty 30
+        thresh50 (dr/->Threshold fifty)
+        temp50 (->Temperature fifty mci)
+        temp40 (->Temperature forty lax)
+        temp30 (->Temperature thirty san)
+        temp20 (->Temperature twenty chi)
+        ws50 (->WindSpeed fifty mci)
+        ws40 (->WindSpeed forty lax)
+        ws10 (->WindSpeed ten irk)
+        fired (-> s
+                  (insert thresh50
+                          temp50
+                          temp40
+                          temp30
+                          temp20
+                          ws50
+                          ws40
+                          ws10)
+                  fire-rules)
+
+        unpaired-res (query fired dr/unpaired-wind-speed)
+        cold-res (query fired dr/cold-temp)
+        hot-res (query fired dr/hot-temp)
+        temp-his-res (query fired dr/temp-his)
+        temps-under-thresh-res (query fired dr/temps-under-thresh)
+
+        create-serializer (fn [stream]
+                            ;; Currently only one.
+                            (condp = serde-type
+                              :fressian (df/create-session-serializer stream)))
+
+        rulebase-baos (java.io.ByteArrayOutputStream.)
+        rulebase-serializer (create-serializer rulebase-baos)
+
+        session-baos (java.io.ByteArrayOutputStream.)
+        session-serializer (create-serializer session-baos)
+
+        holder (atom [])
+        mem-serializer (->LocalMemorySerializer holder)]
+
+    ;; Serialize the data.  Store the rulebase seperately.  This is likely to be the most common usage.
+    
+    (d/serialize-rulebase fired
+                          rulebase-serializer)
+    (d/serialize-session-state fired
+                               session-serializer
+                               mem-serializer)
+
+    (let [rulebase-data (.toByteArray rulebase-baos)
```

Comment:
@rbrush 
This is the transition to byte array backed streams instead of temp files as you commented on previously.